bin/trin-execution/src/e2hs/manager.rs

```diff
@@ -60,7 +60,7 @@ impl E2HSManager {
 
     pub async fn last_fetched_block(&self) -> anyhow::Result<&ProcessedBlock> {
         let Some(current_e2hs) = &self.current_e2hs else {
-            panic!("current_e2hs should always be present, perhaps it wasn't initialized in E2HSManager::new()?");
+            bail!("No blocks were fetched yet, current_e2hs is initialized in get_next_block");
```

Comment:
```suggestion
            bail!("No blocks were fetched yet, last_fetched_block is available only after successful get_next_block");
```
nit: I would phrase it like this. Up to you.